src/main/java/hlf/java/rest/client/util/XssProtectionUtil.java

```diff
@@ -0,0 +1,34 @@
+package hlf.java.rest.client.util;
+
+import static hlf.java.rest.client.exception.ErrorCode.XSS_VALIDATION_FAILED;
+import static org.apache.commons.lang3.StringEscapeUtils.escapeHtml4;
+
+import hlf.java.rest.client.exception.InputValidationException;
+import java.io.IOException;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@UtilityClass
+public class XssProtectionUtil {
+
+  public String validateAndGetXssSafeString(String inputArg) {
+    if (!escapeHtml4(inputArg).equals(inputArg)) {
+      throw new InputValidationException(
+          XSS_VALIDATION_FAILED,
+          "Xss validation failed, parameter: " + inputArg + " may contain Xss strings");
```

Comment:
[nitpick] XSS is an abbreviation, it should be in all caps. 
Is it a good idea to pass the input argument, that will give hint. We shouldn't leak out such details.